eclipse-collections-code-generator/src/main/resources/api/map/immutableObjectPrimitiveMap.stg

```diff
@@ -39,22 +38,33 @@ public interface ImmutableObject<name>Map\<K> extends Object<name>Map\<K>
     @Override
     Immutable<name>Collection reject(<name>Predicate predicate);
 
-    /**
-     * @since 9.0.
-     */
-    default ImmutableObject<name>Map\<K> tap(<name>Procedure procedure)
```

Comment:
Hi @jdimeo, why is this method being removed?

Reply:
@donraab it's redundant with/trivially overrides `tap()` that is also declared in the parent interface `objectPrimitiveMap`.

Reply:
Is the type that is returned here different then in the parent interface? If so, this is a breaking change. This should also not be grouped together with the great JavaDoc work you are doing here as it is unrelated. 

Please revert it and if it is truly a duplicate we can remove it in a separate PR.

Reply:
@donraab sounds good I will revert (pending other feedback). Even if the return type is changing, I think we should implement it by calling the super default impl with an `@Override` instead of reimplementing

Reply:
Thanks @jdimeo. We've had some discussion on how best to address issues like this. While I agree that calling super is conceptually the better thing to do, in practice for a single method call implementation like forEach, it winds up being less verbose duplicating the method call. This is non-obvious at first. I had to try it to see myself. If there is more than a single method call then calling super is definitely the only reasonable option. It bothers me to have even a single line of duplicate code, but this is a case where I've learned to deal with it.

Reply:
Thanks @donraab - I see it differently, because if down the road if I change the default implementation in the super interface, I would expect all subclasses/sub-interfaces to use my new implementation (unless they meaningfully overrode it) and not want to have to track down every duplicated re-implementation. Verbosity (or lack thereof) isn't at all what I was going for. But of _course_ I'm happy to defer to you all as maintainers of this code base!

Reply:
We see things quite similarly I think. I've just allowed myself some flexibility in the cases of single method default methods where the likelihood of there ever being a change is as close to zero as possible, and the cost in verbosity or worse, casting is non zero. I sometimes save my argument points. I'll see if I can dig up the discussion thread we recently had on this and I'll link it here and extract the approach into some docs.

Reply:
@jdimeo @motlin @nikhilnanivadekar Here's the recent discussion on this topic of one line duplication vs. super calls for reference here: https://github.com/eclipse/eclipse-collections/pull/862#pullrequestreview-392463207

Reply:
@jdimeo I will try and read through all of the Javadoc additions in your PR by the weekend. Thank you for this terrific contribution, it is much appreciated.

Reply:
Thanks @donraab - no rush as it took me over 3 years to do this much ;-) you and others have been very responsive. thank you!